compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Archive.hs

```diff
@@ -69,6 +71,11 @@ encodeArchiveLazy package =
           }
     in Proto.toLazyByteString archive
 
+encodePackageHash :: LF.Package -> T.Text
```

Comment:
As @cocreature suggested, `encodeArchiveLazy` should return the hash. When we do this, we should also introduce a newtype for hashes. It's easy to get lost in all these different types of text.